TFInstall.sh

```diff
@@ -1,53 +1,66 @@
 # Check if gnupg is installed
-if (which gpg)
+if (command -v gpg > /dev/null)
 then
   echo "gnupg is already installed."
 else
-  echo "Installing gnupg"
+  echo "Installing gnupg..."
   sudo apt-get install -y gnupg
 fi
 
 # Check if software-properties-common is installed
-if dpkg -s software-properties-common > /dev/null 2>&1; then
-  echo 'software-properties-common is already installed'
+if (dpkg -s software-properties-common > /dev/null 2>&1)
+then
+  echo "software-properties-common is already installed."
 else
-  echo 'Installing software-properties-common...'
+  echo "Installing software-properties-common..."
   sudo apt-get install -y software-properties-common
 fi
 
 # Check if the HashiCorp GPG key and repository are added
-if apt-cache policy terraform | grep -q 'hashicorp'
+if (apt-cache policy terraform | grep -q 'hashicorp')
 then
   echo "HashiCorp repository is already present. Skipping addition of GPG key and repository."
 else
   echo "Adding the HashiCorp GPG key and repository."
 fi
 
-# Check if /etc/apt/keyrings directory exists
-if [ -d "/etc/apt/keyrings" ]
+# Ensure /etc/apt/keyrings directory exists
+if [ -f /etc/apt/sources.list.d/hashicorp.list ]
```

Comment:
I would stick to subshells with exit code execution instead of these implicit test functions. tl;dr it's more clear to stick with running commands like this `if (some command)`